src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1866,12 +1866,9 @@ private long EnforceConfigurationFor(DocumentsOperationContext context, string i
                 var lastModifiedTicks = _database.Time.GetUtcNow().Ticks;
 
                 var local = _documentsStorage.GetDocumentOrTombstone(context, lowerId, throwOnConflict: false);
-                var deletedDoc = local.Document == null && local.Tombstone != null;
-                Debug.Assert(local.Document != null || local.Tombstone != null);
+                var deletedDoc = local.Document == null;
 
-                var docFlags = deletedDoc ? local.Tombstone.Flags : local.Document.Flags;
```

Comment:
what if tombstone is not deleted? shouldn't we take the flags from there?

Reply:
No, it will do the same here if we will pass the doc/tombstone flags or if we will use the `DocumentFlags.None`.